protocol/v2/ssv/validator/committee.go

```diff
@@ -98,12 +98,12 @@ func (c *Committee) StartDuty(logger *zap.Logger, duty *spectypes.CommitteeDuty)
 		return errors.New(fmt.Sprintf("CommitteeRunner for slot %d already exists", duty.Slot))
 	}
 
-	var sharesCopy = make(map[phase0.ValidatorIndex]*spectypes.Share, len(c.Shares))
-	for k, v := range c.Shares {
-		sharesCopy[k] = v
+	validatorShares := make(map[phase0.ValidatorIndex]*spectypes.Share, len(duty.BeaconDuties))
+	for _, bd := range duty.BeaconDuties {
+		validatorShares[bd.ValidatorIndex] = c.Shares[bd.ValidatorIndex]
```

Comment:
should probably add a safety check that if we're missing one of the shares then to return an error (or less risky to just log an error & continue)

Reply:
although maybe we should remove the duty, log an error & continue instead

Reply:
we delete when we call `onShareStop` but it might be right after we sent a duty. so its best to remove the duty like you said.

Reply:
@moshe-blox please take a look on recent commit